### ISSUE TITLE
High Battery Drain in Standby Mode with PlayIntegrityFix v18.1

### ISSUE THREAD

**Issue by hang245141253** [chiteroman/PlayIntegrityFix#498]
Hello, I've noticed a significant battery drain when using PlayIntegrityFix v18.1. Specifically, when the module is active, my phone's standby power consumption increases dramatically. Without the module enabled, the overnight drain is around 4%, but with the module, it jumps to approximately 16%.

Steps to Reproduce:

Install PlayIntegrityFix v18.1 on the device.
Enable the module.
Leave the phone on standby overnight.
Observe the battery consumption difference between enabling and disabling the module.
Environment Details:

Device: Xiaomi MI 10T Pro
OS Version: MIUI 14 (EU Version)
Module Version: PlayIntegrityFix v18.1
Additional Notes:
I conducted several tests over multiple nights to ensure the results are consistent. This issue does not occur when the module is not active.

Please let me know if you need additional information or logs to assist with debugging. Thank you for your work on this project, and I hope this issue can be resolved soon.

**Comment by chiteroman** [comment]
[PlayIntegrityFix_v18.2.1-EXPERIMENTAL.zip](https://github.com/user-attachments/files/18411738/PlayIntegrityFix_v18.2.1-EXPERIMENTAL.zip)

Can you try this version?

**Comment by mohitjalan07** [comment]
Yeah, I faced the same, overnight drain is quite High when pif is installed, it drains like 10-15% battery on standby, but while using rom's built-in play integrity spoof it hardly drains 5%

**Comment by rmuxnet** [comment]
I can confirm too battery drain went from 1% to 14% on idle 

info : 
Device: Samsung Galaxy A71
OS Version:A14 Crdroid
Module Version: PlayIntegrityFix v18.1

**Comment by CrisPH7** [comment]
Same here, I noticed that my phone was draining more at night, so I replaced it with the Fork (by Osmosis & Chiteroman), so I had no more battery drain 

Device: Redmi 10
OS Version: HyperOS A14
Module Version: PlayIntegrityFix v18.1

**Comment by deckultra734** [comment]
I noticed a similar problem, overnight it unusually drained about 14%. 

Device: OnePlus 11
OS version: LineageOS for MicroG 21 (A14)
Module Version: Play Integrity Fix 18.2

**Comment by UzairKhan458** [comment]
Same issue I'm getting on Oneplus8, Idle drain of battery is too high after using this.

**Comment by Vojtak42** [comment]
14%💀. My phone drains like 40% overnight when it's not on charger. I'm using Fork.

**Comment by chiteroman** [comment]
Anyone here can test v18.3 ?

**Comment by deckultra734** [comment]
> Anyone here can test v18.3 ?

I'll help test.

**Comment by nutzboi** [comment]
does it show in the settings that google play services is the app draining the battery? because i had it draining 40% like vojtak and i blamed google for it and degoogled my phone actually.

**Comment by rmuxnet** [comment]
> > I can confirm too battery drain went from 1% to 14% on idle
> > info : Device: Samsung Galaxy A71 OS Version:A14 Crdroid Module Version: PlayIntegrityFix v18.1
> 
> Did u tried remove module, may ROM problem

not a rom issue .

**Comment by rmuxnet** [comment]
> does it show in the settings' "battery usage" that Google Play Services is the app draining the battery? because i had it draining 40% like vojtak so i blamed google for it and degoogled my phone actually.

yes it did show that . 
(ps i dont have the screen show how much it drained but it did drain 14-20% an hour with screen off) 

![Image](https://github.com/user-attachments/assets/704aa4e4-a20a-4214-b07e-e24f53908677)

**Comment by Kuszki** [comment]
After update I noticed a bit lower battery usage, especially over the night.

**Comment by SteamDeckinson** [comment]
It's a little better on v18.3 compared to 18.2.

18.2 it drained from 20% to 30% overnight, 18.3 is about 10% which is better but not great either, I remember at one point it was around 5% overnight which was around 8 to 9 hours.

Same phone and custom ROM as well.

**Comment by UzairKhan458** [comment]
High battery Drain over the night using Play Integrity 18.6

**Comment by Kuszki** [comment]
After updating to 18.7 battery usage dropped to ~1% overnight.
